### DISCUSSION TITLE
Kong prometheus SHM memory keeps on increasing

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#8464] [Help] [answered]
Prometheus plugin SHM memory usage keeps on increasing. Any way to add TTL here or am i missing some config ?

**Comment by hbagdi** [comment]
How much increase are you observing and across what time durations? It should increase with cardinality but not in an unbounded fashion.

**Reply by shettyh** [reply]
![Screenshot 2022-03-02 at 10 41 02 AM](https://user-images.githubusercontent.com/13759916/156299339-9398080e-5a6e-40ff-bd04-f104a57408d8.png)

@hbagdi thanks for the response. Behaviour is not consistent regarding the memory growth, also once it reaches lets say 80% it wont reduce after that, need to restart pods. What are the factors that effect this kind of memory growth ?

**Comment by ADD-SP** [comment, marked as answer]
The amount of shared memory does not decrease due to the memory management mechanism of the operating system, but this is only an illusion.

The operating system gradually allocates the shared memory to the program according to its needs until it reaches the limit set by the program, for example, 5m.

However, the operating system does not release the allocated shared memory, even if it has not been used for a long time, but the application can use it at any time.

It is due to all this that it appears that the amount of shared memory keeps increasing, but in reality, there may be a large amount of free space.

**Reply by ADD-SP** [reply]
Also, when shared memory is running low, it will automatically free memory according to the LRU policy, and if this is a serious problem, the right thing to do is to increase the size of shared memory, the default size of `prometheus_metrics` is `5m`.
